flow/size.yaml

```diff
@@ -62,3 +83,111 @@ components:
           type: integer
           default: 1518
           x-field-uid: 2
+    Flow.SizeWeightPairs:
+      type: object
+      description: |-
+        Frame size distribution, defined as <size, weight> pairs (including IMIX distribution).
+      properties:
+        choice:
+          type: string
+          default: predefined
+          x-field-uid: 1
+          x-enum:
+            predefined:
+              x-field-uid: 1
+            custom:
+              x-field-uid: 2
+        predefined:
+          type: string
+          default: imix
+          description: |-
+            Specify predefined frame size distribution <size, weight> pairs (including IMIX distribution).  
+            The available predefined distribution pairs are:
+            - IMIX (64:7, 570:4, and 1518:1)  
+            - IPSec IMIX (90:58.67, 92:2, 594:23.66 and 1418:15.67)  
+            - IPv6 IMIX (60:58.67, 496:2, 594:23.66 and 1518:15.67)  
+            - Standard IMIX (58:58.67, 62:2, 594:23.66 and 1518:15.67)  
+            - TCP IMIX (90:58.67, 92:2, 594:23.66 and 1518:15.67)  
+            - RPR Trimodal (64:60, 512:20 and 1518:20)  
+            - RPR Quadmodal (64:50, 512:15, 1518:15 and 9216:20)  
+            - Cisco (64:7, 594:4 and 1518:1)  
+            - Tolly (64:55, 78:5, 576:17 and 1518:23)  
+          x-field-uid: 2
+          x-enum:
+            imix:
+              x-field-uid: 1
+            ipsec_imix:
+              x-field-uid: 2
+            ipv6_imix:
+              x-field-uid: 3
+            standard_imix:
+              x-field-uid: 4
+            tcp_imix:
+              x-field-uid: 5
+            rpr_trimodal:
+              x-field-uid: 6
+            rpr_quadmodal:
+              x-field-uid: 7
+            cisco:
+              x-field-uid: 8
+            tolly:
+              x-field-uid: 9
+        custom:
+          type: array
+          items:
+            $ref: '#/components/schemas/Flow.SizeWeightPairs.Custom'
+          x-field-uid: 3
+    Flow.SizeWeightPairs.Custom:
+      type: object
+      description: |-
+        Custom frame size distribution <size, weight> pair.
+      properties:
+        size:
+          description: The size of the frame (in bytes) for this weight pair.
+          type: integer
+          format: uint32
+          minimum: 64
+          maximum: 65535
+          default: 64
+          x-field-uid: 1
+        weight:
+          description: |-
+            A user-defined, 32-bit integer (weight) assigned to the corresponding frame size in this weight pair. 
+            Higher weight means more packets.
+          type: integer
```

Comment:
Shouldn't weight be float with minimum=0 to allow more distributions ?

Moreover, having more weight means packets will be proportionately higher. I think we need to document how the weights should be interpreted and what kind of distribution is expected (provided example on a separate comment)

Reply:
done